File: 人机交互2020/9-4/1-2 VB.net+类/Test/clsDir.vb
Imports System.IO

Public Class clsDir
    Private files() As String
    Private dirs() As String

    Public Sub New(ByVal path As String)
        files = Directory.GetFiles(path)
        dirs = Directory.GetDirectories(path)
    End Sub
    Public Sub Output(ByVal lstInfo As ListBox)
        For i = 0 To files.Count - 1
            lstInfo.Items.Add(files(i))
        Next
        For i = 0 To dirs.Count - 1
            lstInfo.Items.Add(dirs(i))
        Next
    End Sub

End Class

File: 人机交互2020/9-4/9-4/2-2 C#+类/Test/clsDir.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.IO;
using System.Windows.Forms;

namespace Test
{
    class clsDir
    {
        private String[] Dirs;
        private String[] Files;
        public clsDir(String path)
        {
            Dirs = Directory.GetDirectories(path);
            Files = Directory.GetFiles(path);
        }
        public void  Output(ListBox lstInfo)
        {
            lstInfo.Items.Clear();
            for (int i = 0; i < Dirs.Count(); i++)
                lstInfo.Items.Add(Dirs[i]);
            for (int i = 0; i < Files.Count(); i++)
                lstInfo.Items.Add(Files[i]);
        }
    }
}
